doomworld_downloader/lmp_parser.py

```diff
@@ -8,3 +8,67 @@
     - Add a property to upload.ini that is points to the path of the parse_lmp script
   - Your output should be a dictionary of all relevant info for the final JSON
 """
+
+import logging
+import subprocess
+
+from .upload_config import CONFIG
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+class LMPData:
```

Comment:
Do we need a data class or do we just need a parser (get the data and send the relevant parts to the manager)?

Reply:
Good question; I hadn't thought how this will interact with the data manager yet because I was planning on getting all of these classes for parsing every bit of info out, so I can see clearly what data we have per class more clearly before figuring out how to move them all together.

An idea that just came to me is that each of these classes could have their own data manager perhaps, that will just store the data, then the data manager could have an intersect function? Not sure if that's better, I guess it depends on the above comment as to whether it's even possible for the data manager to play that role or if we want an intermediate class that does that.

Reply:
I think it already handles the intersect alright with basically the "message passing" interface. An intermediate object makes sense to me if there ends up needing to be more work done to massage the data before it gets into the manager. Like if we want to just take raw data and put it into a lmp data object, pass that to something else that does some interpretation _before_ sending the final stuff to the data manager. But that interpretation is kind of an internal detail of the parsing. It could either be in the parser or the parser sends data to the interpreter and it then sends stuff to the data manager. I guess it would be a matter of how big the files get and what ends up looking cleaner. I'm certainly a fan of lots of small files with limited concerns :wink:

Reply:
Yeah, putting some stuff in the parser makes sense, but there is possibly going to be a lot of custom logic for some of the fields that depends on all the sources, which is why an intermediate class might make sense. Yeah, I like the data manager as it is right now since it's small and to-the-point, and adding a bunch of very custom logic to it will make it very ugly. :D Better keep that hidden away in some intermediate modules no one will look at if we need it ;)

Reply:
I can try to update this to be a parser and engage the data manager and see what that looks like. Might try to make a separate PR to finalize the main script before doing that

Reply:
Ya, ultimately we'll just have to go in a direction and make changes when we hit a wall :^)